rocks/RocksTransaction.java

```diff
@@ -239,23 +239,18 @@ void closeStorage() {
     }
 
     public static class Data extends RocksTransaction {
-
-        private final SchemaGraph schemaGraph;
         private final RocksStorage.Data dataStorage;
+        private final RocksDatabase.Cache cache;
 
         public Data(RocksSession.Data session, Arguments.Transaction.Type type, Options.Transaction options) {
             super(session, type, options);
 
-            long lock = session.database.dataReadSchemaLock().readLock();
-            schemaGraph = session.database.cache().schemaGraph();
-            schemaGraph.incrementReference();
-            session.database.dataReadSchemaLock().unlockRead(lock);
-
+            cache = session.database.borrowCache();
```

Comment:
we no longer to lock with the read lock since the access to the cache and the process of incrementing the reference has been wrapped into a synchronised method `borrowCache`

Reply:
According to @lriuui0x0, volatile is not needed in the context of explicit synchronization: https://stackoverflow.com/a/32974360/1622847